JOSE_paper/paper.md

```diff
@@ -0,0 +1,254 @@
+---
+title: 'Research Software Engineering Skills'
+tags:
+  - Software Engineering
+  - Python
+  - Research Software
+authors:
+  - name: Jack Atkinson
+    orcid: 0000-0001-5001-4812
+    affiliation: "1"
+    corresponding: true
+  - name: Amy Pike
+    affiliation: "1"
+    orcid: 0009-0007-7278-4122
+  - name: Marion Weinzierl
+    affiliation: "1"
+    orcid: 0000-0002-2302-5476
+affiliations:
+ - name: Institute of Computing for Climate Science, University of Cambridge, UK
+   index: 1
+date: 29 July 2024
+bibliography: paper.bib
+
+---
+
+# Summary
+
+Research across many domains increasingly relies on software for simulation,
+data processing, analysis, visualisation, and more.
+This software is often written by individuals without a formal training in software engineering
+leading to inefficiencies, bugs, and poor reproducibility/re-usability.
+
+The course described in this paper is designed to raise awareness software
+engineering principles and how they can be applied to research code.
+It illustrates, through a series of practical exercises, why these are important and
+introduced basic techniques to produce FAIR-er code [@barker2022introducing].
+We hope that introducing a software engineering mindset, with simple tools and techniques
+to facilitate this will have long-term impact with participants continuing to explore
+ideas in their future work beyond this course.
+
+
+
+# Statement of need
+
+Software forms a significant part of research today, with much of it being written and developed
+by academic researchers. Many of these individuals are self-taught or have minimal training,
+which will often focus on 'coding' rather than software development.
+As a result there is often a lack of awareness of software development principles
+amongst academic developers leading to software that is hard to use and maintain,
+presenting a barrier to research.
+
+In our experience working with academic developers many of the bad practices and
+anti-patterns are a result of them:
+
+- learning on the job from old code and emulating habits they observe
+- never having been made aware of the available tools and techniques
+- not keeping up to date with language/package developments
+- not investing time into writing good code, often due to placing priorities elsewhere.
+
+The intent of developing this short course, _RSE Skills (in Python)_, is to educate
+individuals in some basic software engineering principles that they can carry forward
+in their work.
+It is designed to introduce key tools and techniques to facilitate the writing of
+better code, but also to raise awareness of research software engineering and provide pointers
+to allow researchers to continue exploring and learning beyond this material.
```

Comment:
I am wondering whether we should link to https://carpentries-lab.github.io/good-enough-practices/ and/or the corresponding https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1005510 here , and explain how our course/paper is similar and how it is different.

Reply:
I would be happy to if you think it is a good idea.
I'm not super familiar with the work so would you be able to add something?

Reply:
OK, I just spent way too much time to write a couple of sentences about this here: https://github.com/jatkinson1000/rse-skills-workshop/pull/14/commits/0030a73f53baf7dc444470d940693a5ea3d08646